mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -282,17 +297,33 @@ public MessageMetaData copy(Mailbox mailbox, MailboxMessage original) throws Mai
                 imapUidDAO.insert(composedMessageIdWithMetaData));
     }
 
-    private CompletableFuture<Void> manageUnseenMessageCounts(Mailbox mailbox, Flags oldFlags, Flags newFlags) {
+    private CompletableFuture<Void> manageCounters(Mailbox mailbox, UpdatedFlags updatedFlags) {
+        return CompletableFuture.allOf(manageUnseenMessageCounts(mailbox, updatedFlags),
+                manageRecents(mailbox, updatedFlags));
+    }
+
+    private CompletableFuture<Void> manageUnseenMessageCounts(Mailbox mailbox, UpdatedFlags updatedFlags) {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
-        if (oldFlags.contains(Flag.SEEN) && !newFlags.contains(Flag.SEEN)) {
+        if (updatedFlags.isUnset(Flag.SEEN)) {
             return mailboxCounterDAO.incrementUnseen(mailboxId);
         }
-        if (!oldFlags.contains(Flag.SEEN) && newFlags.contains(Flag.SEEN)) {
+        if (updatedFlags.isSet(Flag.SEEN)) {
             return mailboxCounterDAO.decrementUnseen(mailboxId);
         }
         return CompletableFuture.completedFuture(null);
     }
 
+    private CompletableFuture<Void> manageRecents(Mailbox mailbox, UpdatedFlags updatedFlags) {
```

Comment:
looks like a pure logic duplication.
You probably miss a level of abstraction to be DRY